crates/dprint/src/cli/run_cli.rs

```diff
@@ -138,6 +138,16 @@ fn get_file_paths_by_plugin(plugins: &Vec<Box<dyn Plugin>>, file_paths: Vec<Path
                 } else {
                     file_paths_by_plugin.insert(String::from(plugin.name()), vec![file_path]);
                 }
+                continue;
+            }
+        }
+        if let Some(file_fullname) = file_path.file_name().and_then(|s| s.to_str()).map(|s| s.to_string()) {
+            if let Some(plugin) = plugins.iter().filter(|p| p.file_fullnames().contains(&file_fullname)).next() {
+                if let Some(file_paths) = file_paths_by_plugin.get_mut(plugin.name()) {
+                    file_paths.push(file_path);
+                } else {
+                    file_paths_by_plugin.insert(String::from(plugin.name()), vec![file_path]);
+                }
             }
```

Comment:
I think we should change this matching to prioritize plugins based on the order provided in the configuration file (so the order of `plugins` here). We could loop over file paths first, then plugins, then check the file extension and then the file path. This would allow people to specify a plugin like dprint-plugin-cargo-toml first and then a future plugin like dprint-plugin-toml... so it would format Cargo.toml with dprint-plugin-cargo-toml and then other .toml files with dprint-plugin-toml.

Alternatively, perhaps it should match based on the exact file name first for all plugins and then match based on the file extension.

Thoughts on what would be best?